### PR TITLE
Adds character eye customization #2946

### LINKED ISSUE #2946: You cant customize your character eye color in the customization menu

### PR DESCRIPTION
**Pull request by Rember**
Fixes #2946
This adds a set of color bars in the character customization screen that lets you adjust the eye colors of your character.

There's still a handful of problems:
* The bars don't fit on the default window size, so you'll have to lengthen your window to see them.
* The bars don't stretch out, in order to customize the colors right now you have to use the number input next to the bars.
* All of these functionalities for eye colors were pretty much copy-pasted from hair color code (with some deletions to parts of code that seemed irrelevant), so there's probably a whole mess of redundant or plain wrong code. I'm not really experienced with C# so apologies for that.

Please leave feedback or help if you can! You can also just ping me in the discord server, RemberBL#9516

UPDATE: I'm working on refactoring HumanoidProfileEditor.cs alongside these changes, somebody suggested I might as well do that while I'm doing this.

UPDATE #2: I've been busy with a lot of Important Stuff™ so I've ended up neglecting this PR since the last week. I did manage to refactor some code but it's throwing an exception when the lobby loads in. If anybody wants to give a hand just shoot me a message, I'd really appreciate it.

### CODE REVIEW

File: Content.Client/UserInterface/HumanoidProfileEditor.cs [hunk 1]
```diff
@@ -360,7 +361,33 @@ public HumanoidProfileEditor(IClientPreferencesManager preferencesManager, IProt
...
                 }
 
-                #endregion Clothing
+                #endregion Backpack
+
+                #region Eyes
+
+                {
+                    var panel = HighlightedContainer();
+                    panel.SizeFlagsHorizontal = SizeFlags.None;
```
Comment (Rockdtben) [inline Content.Client/UserInterface/HumanoidProfileEditor.cs:370]: `panel.SizeFlagsHorizontal = SizeFlags.None;`

should be 

`panel.SizeFlagsHorizontal = SizeFlags.FillExpand;`

Like PJB said.

File: Content.Shared/GameObjects/Components/SharedMagicMirrorComponent.cs [hunk 2]
```diff
@@ -42,20 +42,33 @@ public HairColorSelectedMessage((byte r, byte g, byte b) color, bool isFacialHai
...
+        [Serializable, NetSerializable]
+        public class EyeColorSelectedMessage : BoundUserInterfaceMessage
+        {
+            public (byte r, byte g, byte b) EyeColor;
+
+            public EyeColorSelectedMessage((byte r, byte g, byte b) color)
+            {
+                EyeColor = color;
+            }
+        }
```
Comment (PaulRitter) [inline Content.Shared/GameObjects/Components/SharedMagicMirrorComponent.cs:54]: you can pull latest master & just use the color struct here

File: Content.Client/GameObjects/Components/MagicMirrorBoundUserInterface.cs [hunk 3]
```diff
@@ -257,10 +262,139 @@ public ColorSlider(string styleClass)
...
+                        _colorValue = (byte) result;
+                        _slider.Value = result;
+                        _ignoreEvents = false;
+
+                        OnValueChanged?.Invoke();
+                    }
+                };
+            }
+        }
+    }
```
Comment (PaulRitter) [inline Content.Client/GameObjects/Components/MagicMirrorBoundUserInterface.cs:391]: might as well use xaml to create a new control but i wont force you :P

**Review by PaulRitter — CHANGES_REQUESTED** [review]
solid pr :+1:

**Comment by metalgearsloth** [comment]
@Rember were you coming back to this?

**Comment by Rember** [comment]
> 
> 
> @Rember were you coming back to this?

I got caught up in a lot of stuff lately, but I do want to come back to it (if it hasn't run too far behind master). I'll see if I can get back on it around this time tomorrow, then I'll be looking for help in #howdoicode. Feel free to ping me in case you want to help out with this.

**Comment by metalgearsloth** [comment]
No worries I'll leave it up, was just checking if it was derelict.

**Comment by SweptWasTaken** [comment]
> 
> 
> > @Rember were you coming back to this?
> 
> I got caught up in a lot of stuff lately, but I do want to come back to it (if it hasn't run too far behind master). I'll see if I can get back on it around this time tomorrow, then I'll be looking for help in #howdoicode. Feel free to ping me in case you want to help out with this.

No progress in 14 days, closing.

**Pull request closed without merging**